qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -0,0 +1,273 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from operator import itemgetter
+from os.path import basename
+from collections import defaultdict
+
+from future.utils import viewitems
+
+from qiita_db.util import (get_filetypes, get_files_from_uploads_folders,
+                           get_data_types, convert_to_id, get_filepath_types)
+from qiita_db.study import Study
+from qiita_db.data import RawData
+from qiita_db.ontology import Ontology
+from qiita_db.metadata_template import (PrepTemplate, TARGET_GENE_DATA_TYPES,
+                                        PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+from qiita_db.parameters import (Preprocessed454Params,
+                                 PreprocessedIlluminaParams)
+from qiita_pet.util import STATUS_STYLER
+from qiita_pet.handlers.util import download_link_or_path
+from .base_uimodule import BaseUIModule
+
+
+filepath_types = [k.split('_', 1)[1].replace('_', ' ')
+                  for k in get_filepath_types()
+                  if k.startswith('raw_')]
+fp_type_by_ft = defaultdict(
+    lambda: filepath_types, SFF=['sff'], FASTA=['fasta', 'qual'],
+    FASTQ=['barcodes', 'forward seqs', 'reverse seqs'])
+
+
+def get_accessible_raw_data(user):
+    """Retrieves a tuple of raw_data_id and the last study title for that
+    raw_data
+    """
+    d = {}
+    for sid in user.user_studies:
+        for rdid in Study(sid).raw_data():
+            d[int(rdid)] = Study(RawData(rdid).studies[-1]).title
+    return d
+
+
+def get_raw_data(rdis):
+    """Get all raw data objects from a list of raw_data_ids"""
+    return [RawData(rdi) for rdi in rdis]
+
+
+def get_prep_templates(pt_ids):
+    """Get all the prep template objects from a list of ids
+
+    Parameters
+    ----------
+    pt_ids : list of int
+        The prep template ids
+
+    Returns
+    -------
+    list of PrepTemplate
+    """
+    return [PrepTemplate(pt_id) for pt_id in sorted(pt_ids)]
+
+
+class PrepTemplateTab(BaseUIModule):
+    def render(self, study, full_access):
+        files = [f for _, f in get_files_from_uploads_folders(str(study.id))]
+        data_types = sorted(viewitems(get_data_types()), key=itemgetter(1))
+        prep_templates_info = [
```

Comment:
Changed by a generator function. It actually get's rid of the `get_prep_templates` function.